[12.x] Extract repeated Traversable conversion in Str class

mohammadrasoulasghari opened the pull request:
## 🔍 In this PR

In this PR, I've refactored repeated code in the `Str` class by extracting a common pattern into a dedicated helper method. 

The pattern `if ($value instanceof Traversable) { return (new Collection($value))->all(); }` was repeated 5 times across several methods in the `Str` class (`replace`, `replaceArray`, `remove`, etc.). This repetition violated the DRY (Don't Repeat Yourself) principle and made the code harder to maintain.

I've extracted this logic into a private static helper method called `toArrayIfTraversable` which:
- ✅  if it's a `Traversable`, converts it to an array using
- ✅ If not, returns the original value unchanged

### 🌟 Benefits:

- **Improved Readability**: Code is now cleaner and intent is clearer
- **Better Maintainability**: Future changes to this logic need to be made in only one place
- **Single Responsibility Principle**: Each method now focuses on its core functionality without mixing in type conversion logic
- **Performance**: Using `iterator_to_array()` directly is more efficient than creating a Collection object

### 🔄 Changes:

- Added a new private static helper method `toArrayIfTraversable`
- Updated `replace`, `replaceArray`, and `remove` methods to use this helper
- No changes to public API or method signatures
- No breaking changes - all methods maintain exactly the same behavior

This refactoring is purely internal and has no impact on the public API or behavior of the `Str` class. It simply makes the code more maintainable and follows best practices for code organization.

mohammadrasoulasghari committed:
refactor: extract traversable to array conversion to a private helper method

Extract repeated Traversable to array conversion logic into a private
helper method to improve code readability and maintainability.

shaedrich commented on src/Illuminate/Support/Str.php:
```diff
@@ -1219,23 +1217,30 @@ private static function toStringOr($value, $fallback)
+     * Convert a Traversable to an array, or return the original value if not Traversable.
+     *
+     * @param  mixed  $value
+     * @return mixed
```
You can probably enhance is by using a conditional return value type:
```suggestion
     * @return ($value is \Traversable ? array : mixed)
```

mohammadrasoulasghari committed:
docs: improve PHPDoc for toArrayIfTraversable with conditional return type

return type annotation that more accurately describes the method's behavior:
- Returns an array when the input is a Traversable
- Returns the original value unchanged otherwise

Co-authored-by: Sebastian Hädrich <11225821+shaedrich@users.noreply.github.com>

taylorotwell commented:
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If applicable, please consider releasing your code as a package so that the community can still take advantage of your contributions!
